internal/controller/flow_aggregation.go

```diff
@@ -0,0 +1,270 @@
+// Copyright 2025 Illumio, Inc. All Rights Reserved.
+
+package controller
+
+import (
+	"container/list"
+	"context"
+	"fmt"
+	"time"
+
+	pb "github.com/illumio/cloud-operator/api/illumio/cloud/k8sclustersync/v1"
+	"go.uber.org/zap"
+)
+
+type FlowKey struct {
+	// Timestamp is the time the network event occured.
+	Timestamp int64
+	// SrcIP is the source IP address involved in the network event.
+	SrcIP string
+	// DstIP is the destination IP address involved in the network event.
+	DstIP string
+	// SrcPort is the source port number involved in the network event.
+	SrcPort int
+	// DstPort is the destination port number involved in the network event.
+	DstPort int
+	// Proto is the protocol used in the network event (e.g., TCP, UDP).
+	Proto string
+	// SourceEndpoint contains k8s metadata for source endpoint
+	SourceEndpoint string
+	// DestinationEndpoint contains k8s metadata for destination endpoint
+	DestinationEndpoint string
+}
+
+type FlowCache struct {
+	cache      map[FlowKey]*list.Element
+	queue      *list.List
+	bufferSize int
+}
+
+// cacheManagerIndefinitely manages the flow cache by evicting expired flows based on the active timeout,
+// processing new flows, and resetting the timer for the next expiration.
+func (sm *streamManager) cacheManagerIndefinitely(ctx context.Context, logger *zap.Logger) error {
+	const activeTimeout = 20 * time.Second // Define the active timeout period for flows. TODO: make configurable
+
+	// Create a new timer to trigger every activeTimeout duration.
+	timer := time.NewTimer(activeTimeout)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-ctx.Done(): // If the context is canceled, exit the loop and return the error.
+			return ctx.Err()
+
+		case <-timer.C:
+			now := time.Now().UTC()
+			expiredCutoff := now.Add(-activeTimeout) // Determine the cutoff time for expired flows.
+
+			// Iterate through the flow cache queue to evict expired flows.
+			for sm.FlowCache.queue.Len() > 0 {
+				frontElem := sm.FlowCache.queue.Front()
+				var flowTimestamp time.Time
+				var flowKey FlowKey
+				var err error
+
+				// Check the type of flow and get the flow timestamp and flow key.
+				switch f := frontElem.Value.(type) {
+				case *pb.FalcoFlow:
+					flowTimestamp = f.GetTimestamp().AsTime()
+					flowKey, err = sm.createFlowKey(f)
+				case *pb.CiliumFlow:
+					flowTimestamp = f.GetTime().AsTime()
+					flowKey, err = sm.createFlowKey(f)
+				default:
+					break
```

Comment:
That's not right. You should panic in this case.
Or better, define a common struct or interface type to avoid this situation.